Отправляю фронтенд на ревью

BarhatYo opened the pull request:
Отправляю фронтенд на ревью.
Просьба учесть, что не сделана валидация форм и не актуализированы ошибки - это будет сделано на следующем этапе.

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
    - **На 768px не по макету:**
    - https://disk.yandex.com/i/atb6x8UHfBQfpQ
    - https://disk.yandex.com/i/XdcgV6qRyFa9tg
    - **На 320px не по макету:**
    - https://disk.yandex.com/i/SgYxTYI3Y6NfnA
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
    - `Адаптивный сайт` ведёт в пустоту, проверьте публичность репозитория
    - `about-me__portfolio-item` используйте одну ссылку для текста и кнопки, при клике на любую область
    - Кнопка `Выйти из аккаунта` должна возвращать пользователя на главную страницу
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/HcAbqNF50X9N3A

- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
    - проект имеет ошибки валидации, удобно проверить валидатором https://validator.w3.org/#validate_by_input , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/siwaHhXlF2aUBg
    - Контентное содержимое страницы нужно обернуть тегом `<main>` **по всем роутам** (всё кроме `header и  footer`)
    - в роуте `/главная, /signin , /signup, /404, /profile, /movies, /saved-movies` не хватает сематического тега `<main>`
    - `promo, about-me` `<section>`
    - `techs__items about-me__portfolio-item footer__links movies-card-list` однотипные элементы следует группировать с помощью списка `<ul>`
    - `auth-controls` элементы представляют навигацию `<nav>`
    - `not-found__title login__title register__title profile__hello` является заголовком первого уровня
    - `movies-card__name` является заголовком второго уровня
    - Чекбокс относится к форме поиска, его необходимо вложить внутрь тега `<form>` с поисковым инпутом
    - Всем кнопкам в проекте нужен заполненный атрибут `type`, если кнопка **не** отвечает за отправку формы то значение будет `"button"`
    - `profile_authorized.png` текстовое содержимое следует разметить, оно может изменится
- [ ] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
    - `.not-found__button-back` в абсолютном позиционировании нет необходимости, элемент будет лучше оставить в общем потоке, элемент можно расположить по сетке `flex/grid`
- [ ] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
    - Для расположения содержимого страницы(`class="page"`) необходимо использовать сетку flex/grid, например для реализации `sticky footer`
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
    - Всем интерактивным элементам(ссылки, кнопки, чекбоксы, лого) нужен эффект при наведении и выставленный курсор, проконтролируйте данный критерий для всех интерактивных элементов в проекте
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [ ] <!--2.14--> Иконки экспортированы в формате SVG.
    - `profile_authorized.png promo-logo.png link-icon.jpg` необходимо добавить в векторном формате
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
    - Всем инпутам проекта нужен стилизованный фокус и переопределённый `outline`
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
    - Всем инпутам проекта нужно добавить плейсхолдеры
    - Поисковой инпут нужно сделать обязательным к заполнению
    - Всем полям **Имя и Пароль** нужна валидация минимальной/максимальной длинны значения
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **54**


### Доп комментарии

- Неверно указан язык документа

BarhatYo commented:
@AliaksandrHrychonak приветствую.

Подскажи, пожалуйста, оптимальное решение для реализации:
**.not-found__button-back в абсолютном позиционировании нет необходимости, элемент будет лучше оставить в общем потоке, элемент можно расположить по сетке flex/grid**

Надо же сделать так, чтобы ошибка 404 и текст были по центру, а кнопка внизу.
Понятно, что я могу сделать спейс-битвин, а дальше отступами подогнать необходимую позицию, чтобы было, как на макете. Но я хочу реализовать гибкую историю, чтобы на устройстве любой высоты ошибка 404 была по центру, а кнопка прижата к низу.

У наставников уточнял, но кажется, что какие-то не очень оптимальные решения предлагаются, в том числе с позишн фиксед.

Буду очень благодарен за рекомендации.

BarhatYo commented:
Для инфо - вот, что предлагается:
Вариантов полно:
1. Флексом, с помощью align-self отцентрировать 404, кнопка назад либо флексом либо position: fixed 
2. Отцентрировать 404 флексом, назад position: fixed
3. calc(50% - height/2), назад уже как расположишь, тот же fixed 
4. С помощью transform translate 50%

AliaksandrHrychonak reviewed (commented):
# Резюме по работе:

## Верстка и JSX

### Критерии, влияющие на работоспособность

- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
    - проект имеет БЭМ ошибки, удобно проверить валидатором https://nglazov.github.io/bem-validator-page/ , проконтролируйте данный критерий по всем роутам, пример https://disk.yandex.com/i/cJtwNWG57qpRcA
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
    - [x] <!--3.38--> разметка заключена в (), лишние пустые <div> для этого не используются;
    - [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
    - [x] <!--3.38--> директория images/ с картинками,
    - [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
    - [x] <!--3.38--> директория fonts/ со шрифтами.
    
### Хорошие практики

- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации

- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;
    
## Количество баллов: **94**


### Доп комментарии

- Логотипам по всем роутам также будет хорошо добавить эффект наведения
- `profile_authorized.png` следует удалить из проекта

AliaksandrHrychonak commented:
> @AliaksandrHrychonak приветствую.
> 
> Подскажи, пожалуйста, оптимальное решение для реализации: **.not-found__button-back в абсолютном позиционировании нет необходимости, элемент будет лучше оставить в общем потоке, элемент можно расположить по сетке flex/grid**
> 
> Надо же сделать так, чтобы ошибка 404 и текст были по центру, а кнопка внизу. Понятно, что я могу сделать спейс-битвин, а дальше отступами подогнать необходимую позицию, чтобы было, как на макете. Но я хочу реализовать гибкую историю, чтобы на устройстве любой высоты ошибка 404 была по центру, а кнопка прижата к низу.
> 
> У наставников уточнял, но кажется, что какие-то не очень оптимальные решения предлагаются, в том числе с позишн фиксед.
> 
> Буду очень благодарен за рекомендации.


Вариантов действительно много, много использовать реализацию через flex с отступами, или можно использовать grid

Пример реализации:

![2023-12-22_18-26-46](https://github.com/BarhatYo/movies-explorer-frontend/assets/74908494/98c41591-71b7-4882-9245-c835eb6de21b)
![2023-12-22_18-27-06](https://github.com/BarhatYo/movies-explorer-frontend/assets/74908494/886bdc11-31ec-4b3d-8f7f-29470885b387)


Реализация которая сейчас в проекте вполне корректная!

BarhatYo commented:
@AliaksandrHrychonak Спасибо за ревью!

Нашел ошибку по БЭМ в именовании .header__menu-mobile_active.
Поправил на .header__menu_mobile-active.

Это же единственная ошибка? Следует ли запушить и отправить на ревью, чтобы получить 100?

BarhatYo commented:
Логотипам по всем роутам также будет хорошо добавить эффект наведения
profile_authorized.png следует удалить из проекта

Выполнено.
